static/js/public/carousel.js

```diff
@@ -0,0 +1,130 @@
+/* Carousel class */
+class Carousel {
+  /**
+   * Create a carousel instance with a provided 'holder'
+   *
+   * Required html format:
+   * <div class="p-carousel">
+   *   <div class="p-carousel__mask">
+   *     <div class="p-carousel__scroller">
+   *       <div class="p-carousel__item">
+   *         ...
+   *       </div>
+   *       ...
+   *     </div>
+   *   </div>
+   *   <button class="p-carousel__next">Next</button>
+   *   <button class="p-carousel__prev">Previous</button>
+   * </div>
+   *
+   * @param {HTMLElement|string} holder HTMLElement or selector string for the 'p-carousel' element
+   */
+  constructor(holder) {
+    if (typeof holder === undefined) {
+      throw new Error('Holder not defined');
+    }
+
+    if (typeof holder === 'string') {
+      this.holder = document.querySelector(holder);
+    } else {
+      this.holder = holder;
+    }
+
+    this.scroller = this.holder.querySelector('.p-carousel__scroller');
+    this.items = this.holder.querySelectorAll('.p-carousel__item');
+    this.nextButton = this.holder.querySelector('.p-carousel__next');
+    this.prevButton = this.holder.querySelector('.p-carousel__prev');
+
+    this.currentIndex = 0;
+    this.itemWidth = this.items[0].scrollWidth;
+    this.scrollerWidth = (this.itemWidth) * this.items.length;
+    this.visibleItems = Math.ceil(this.holder.scrollWidth / this.itemWidth);
```

Comment:
Calculating this only when Carousel is created may lead to some cases when it will start acting weird after screen is resized (when number of visible items changes).

Not sure if we need to listen on the resize event. Quite possibly it would be enough to recalculate `visibleItems` just before they are used.

Reply:
Done